api/src/test/java/org/openmrs/module/datafilter/impl/api/db/hibernate/PatientLocationLinkingInterceptorTest.java

```diff
@@ -161,15 +161,15 @@ public void onSave_shouldNotLinkPatientsToLocationsIfTheGpIsNotEnabled() {
 		assertEquals(originalCount, getCountOfPatientLocationLinks().intValue());
 		assertEquals(0, getPatientLocations(patient).size());
 	}
-	
+
 	@Test
 	@Transactional(propagation = Propagation.NOT_SUPPORTED)
 	public void onSave_shouldFailIfThereIsNoSessionLocation() {
 		Patient patient = createTestPatient("789");
-		ee.expect(AssertionFailure.class);
+		ee.expect(HibernateSystemException.class);
 		ee.expectMessage(containsString(("Unable to perform beforeTransactionCompletion callback")));
-		ee.expectCause(isA(DAOException.class));
-		
+		ee.expectCause(Matchers.hasProperty("cause", isA(DAOException.class)));
```

Comment:
DAOException cause is now nested within another HibernateException:
HibernateSystemException > HibernateException > DAOException